app/src/main/java/com/nervousfish/nervousfish/activities/WaitActivity.java

```diff
@@ -0,0 +1,151 @@
+package com.nervousfish.nervousfish.activities;
+
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.TextView;
+
+import com.nervousfish.nervousfish.ConstantKeywords;
+import com.nervousfish.nervousfish.R;
+import com.nervousfish.nervousfish.data_objects.Contact;
+import com.nervousfish.nervousfish.modules.pairing.events.NewDataReceivedEvent;
+import com.nervousfish.nervousfish.service_locator.IServiceLocator;
+
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Used to let the Bluetooth-initiating user know that he should wait for his partner
+ * to complete the pairing session.
+ */
+public final class WaitActivity extends Activity {
+    private static final Logger LOGGER = LoggerFactory.getLogger("WaitActivity");
+    private IServiceLocator serviceLocator;
+    private Object dataReceived;
+    private Object tapCombination;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        this.setContentView(R.layout.activity_wait);
+
+        final Intent intent = getIntent();
+        this.serviceLocator = (IServiceLocator) intent.getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR);
+
+        this.serviceLocator.registerToEventBus(this);
+
+        this.dataReceived = intent.getSerializableExtra(ConstantKeywords.DATA_RECEIVED);
+        tapCombination = intent.getSerializableExtra(ConstantKeywords.TAP_DATA);
+
+        LOGGER.info("dataReceived is not null: " + (this.dataReceived != null)
+                + " tapCombination is not null: " + (tapCombination != null));
+
+        final String message = (String) intent.getSerializableExtra(ConstantKeywords.WAIT_MESSAGE);
+        final TextView waitingMessage = (TextView) findViewById(R.id.waiting_message);
+        waitingMessage.setText(message);
+
+        LOGGER.info("WaitActivity created");
+    }
+
+    private void evaluateData() {
+        //TODO: check if when we decrypt the dataReceived with the tapCombination that we get a normal contact
```

Comment:
Can you remove this `TODO:` comment and include it in the issue I requested you to make 😄